From 85e1b21ec63aa8e9b19dc020a67352060580a990 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sat, 19 Sep 2015 01:15:22 -0400
Subject: [PATCH] Never unsafely unload a chunk

If a chunk is in use... it should not be unloaded.
---
 src/main/java/org/bukkit/craftbukkit/CraftWorld.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index f40a4e7..df999a9 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -216,7 +216,7 @@ public class CraftWorld implements World {
 
     public boolean unloadChunkRequest(int x, int z, boolean safe) {
         org.spigotmc.AsyncCatcher.catchOp( "chunk unload"); // Spigot
-        if (safe && isChunkInUse(x, z)) {
+        if (isChunkInUse(x, z)) { // EMC - never unsafely unload a chunk...
             return false;
         }
 
@@ -227,7 +227,7 @@ public class CraftWorld implements World {
 
     public boolean unloadChunk(int x, int z, boolean save, boolean safe) {
         org.spigotmc.AsyncCatcher.catchOp( "chunk unload"); // Spigot
-        if (safe && isChunkInUse(x, z)) {
+        if (isChunkInUse(x, z)) { // EMC - Never unsafely unload a chunk...
             return false;
         }
 
-- 
1.9.1

